DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/README.md

```diff
@@ -0,0 +1,252 @@
+| Software                          | Intel® oneAPI DPC++ Compiler <br> Intel® FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               | Using the `ac_int` data type for basic operations <br> Efficiently using the left shift operation <br> Setting and reading certain bits of an `ac_int` number
+| Time to complete                  | 20 minutes
+
+
+
+## Purpose
+
+This FPGA tutorial shows how to use the `ac_int` data type with some simple examples.
+
+This data type can be used in place of native integer types to generate area efficient and optimized designs for the FPGA. When you have a computation that does not require the full dynamic range of a 32-bit integer, you should replace your `int` variables with `ac_int` variables of the correct, reduced width. For example, if you know that a loop will iterate from 0 to 12, only 4 bits are required.
+
+Please refer to the [oneAPI DPC++ FPGA Optimization Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide/top/optimize-your-design/resource-use/data-types-and-operations/var-prec-fp-sup/adv-disadv-ac-dt.html) to see advantages and limitations of `ac_int` data types.
+
+### Simple Code Example
+
+An `ac_int` number can be defined as follows:
+```cpp
+ac_int<W, S> a;
+```
+Here `W` is the width in bits and `S` is a bool indicating if the number is signed. Signed numbers use the MSB to store the sign bit.
+
+To use the `ac_int` type in your code, you must include the following header:
+
+```cpp
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+```
+Additionally, you must pass the flag `-qactypes` on Linux or `/Qactypes` on Windows to the `dpcpp` command when compiling your SYCL program in order to ensure that the headers are correctly included. In this tutorial, this is done in `src/CMakeLists.txt`.
+
+### Basic Operations and Promotion Rules
+
+When using `ac_int`, you can write addition, multiplication, and division operations to use precisely as many bits as are needed to store the results.
+
+`ac_int` automatically promotes the result of all operations to the number of bits needed to represent all possible results without overflowing.
+
+For example, the addition of two 8-bit integers results in a 9-bit result to support overflow. Internally, the result will be 9-bit. However, if the user attempts to store the result in an 8-bit container, `ac_int` will let the user do this, which leads to the most significant bit being discarded. The responsibility lies on the user to use the correct data type.
+
+These promotion rules are consistent across all architectures, so the behavior will be equivalent on x86 or on FPGA.
+
+### Shift Operations
+
+The behavior of shift operations in `ac_int` are slightly different from its behavior with native integer types. For full details, see the Algorithmic C (AC) Datatypes documentation available at https://hlslibs.org/. Some key points to remember are as follows:
```

Comment:
1. That is a bold assumption. It's true that AC data types libraries are the only portion we've leveraged, but it doesn't imply all other hlslibs libraries are available.
2. Let's discuss this in Wednesday morning's meeting. Since the [oneAPI FPGA Optimization Guide](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide/top/optimize-your-design/resource-use/data-types-and-operations/var-prec-fp-sup/declare-and-use-the-ac-data-types/declare-the-ac-int-data-type.html) is also referring to that hlslibs website, we should come up with a solution that coordinates this tutorial and the documentation.